packages/core/src/tracing/instrumentation/cloud/aws/lambda.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const cls = require('../../../cls');
+const { EXIT, isExitSpan } = require('../../../constants');
+const tracingUtil = require('../../../tracingUtil');
+const { InstanaAWSProduct } = require('./instana_aws_product');
+
+const MAX_CONTEXT_SIZE = 3582;
+
+const SPAN_NAME = 'aws.lambda.invoke';
+const operations = ['invoke', 'invokeAsync'];
+
+class InstanaAWSLambda extends InstanaAWSProduct {
+  instrumentedMakeRequest(ctx, originalMakeRequest, originalArgs) {
+    const parentSpan = cls.getCurrentSpan();
+    const self = this;
+
+    if (!parentSpan || isExitSpan(parentSpan)) {
+      return originalMakeRequest.apply(ctx, originalArgs);
+    }
+
+    return cls.ns.runAndReturn(() => {
+      const span = cls.startSpan(SPAN_NAME, EXIT);
+      span.ts = Date.now();
+      span.stack = tracingUtil.getStackTrace(this.instrumentedMakeRequest, 1);
+      span.data[this.spanName] = this.buildSpanData(originalArgs[0], originalArgs[1]);
+
+      // eslint-disable-next-line
+      /** @type {import('aws-sdk').Lambda.Types.InvocationRequest | import('aws-sdk').Lambda.Types.InvokeAsyncRequest} */
+      const params = originalArgs[1];
+      let clientContextContentBase64;
+      let clientContextContentJSON;
+      let isJSON = true;
+
+      const instanaCustomHeaders = {
+        Custom: {
+          'x-instana-l': cls.tracingSuppressed() ? '0' : '1',
+          'x-instana-s': span.s,
+          'x-instana-t': span.t
+        }
+      };
+
+      // invokeAsync doesn't have the option ClientContext
+      if (originalArgs[0] === 'invoke') {
+        if (params.ClientContext != null) {
+          clientContextContentBase64 = Buffer.from(params.ClientContext, 'base64').toString();
+
+          try {
+            clientContextContentJSON = JSON.parse(clientContextContentBase64);
+            clientContextContentJSON.Custom = instanaCustomHeaders.Custom;
```

Comment:
What if the existing client context already has the attribute `Custom`? Apparently this is some kind of "standard" attribute that others could be using as well (https://docs.aws.amazon.com/lambda/latest/dg/nodejs-context.html). If I read this correctly, we would  overwrite the existing `Custom` value unconditionally?

I think we need to check if `clientContextContentJSON.Custom` already exists and add our attributes to it if it does.